internal/provider/resource_deployment.go

```diff
@@ -30,6 +30,12 @@ func resourceDeployment() *schema.Resource {
 				Required:    true,
```

Comment:
this should be computed and not required and maybe the name should be node_id to match other resources?